src/main/java/com/puppycrawl/tools/checkstyle/utils/CheckUtil.java

```diff
@@ -680,4 +680,14 @@ public static String getShortNameOfAnonInnerClass(DetailAST literalNewAst) {
         final DetailAST firstChild = parentAst.getFirstChild();
         return extractQualifiedName(firstChild);
     }
+
+    /**
+     * Checks if the given file path is a package-info.java file.
+     *
+     * @param filePath path to the file.
+     * @return true if the package file.
+     */
+    public static boolean isPackageInfo(String filePath) {
+        return filePath.endsWith("package-info.java");
```

Comment:
Won't "my-great-package-info.java" return true from this method?

Reply:
I do not think it is good, but it was like this, so better to fix it in separate issue/PR.
@strkkk , if you agree, please create issue on this.

Reply:
It is a very rare case, but anyway it can be fixed
https://github.com/checkstyle/checkstyle/issues/11531

Reply:
Resolved.